File: back/src/main/java/com/demo/back/campaign/Campaign.java
package com.demo.back.campaign;

import java.util.Collection;
import java.util.UUID;
import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import lombok.Data;

/**
 * Campaign jpa entity.
 */
@Data
@Entity
public class Campaign {

  @Id
  @Column(length = 16)
  @GeneratedValue(generator = "UUID")
  private UUID id;
  private String name;
  @Column(columnDefinition = "TEXT")
  private String description;
  private String imageUrl;
  private boolean actual;

  @OneToMany(mappedBy = "campaign", cascade = {CascadeType.ALL})
  private Collection<Summary> summaries;
}

File: front/app/api/campaign.ts
import {gql} from 'graphql-request'

export type Campaign = {
  id: string
  name: string
  description: string
  imageUrl: string
  actual: boolean
  summaries: Summary[]
}

export type Summary = {
  id: string
  body: string
}

export const getActualCampaign = /* GraphQL */ `
  query getActualCampaign {
    actualCampaign {
      id
      name
      description
      imageUrl
      summaries {
        id
        body
      }
    }
  }
`
